src/detect.c

```diff
@@ -1523,6 +1549,10 @@ static void DetectRunTx(ThreadVars *tv,
 
             StoreDetectFlags(&tx, flow_flags, ipproto, alproto, new_detect_flags);
         }
+        if (new_detect_flags & APP_LAYER_TX_INSPECTED_FLAG) {
```

Comment:
this will only trigger when the tx is complete I think, which means with very large bodies we'll wait too long. I think we need to check per inspection round something like
```c
if (det_ctx->http_body_progress_updated) {
   body->body_inspected = det_ctx->http_body_progress;
   body->http_body_progress_updated = false;
}
```
`det_ctx::http_body_progress_updated` should then only be set if we actually did body inspection (so set by `DetectEngineInspectBufferHttpRequest` and friends after the various limits have been checked)

Reply:
> we need to check per inspection round something like
This means after each call to `DetectRunTxInspectRule`?

Reply:
No I think at the end of each tx iteration in `DetectRunTx`